todo/src/store/reducers/reducer.js

```diff
@@ -0,0 +1,72 @@
+import { UPDATE_TITLE, ADD_NEW_TODO, TOGGLE_TODO, DELETE_TODO } from "../actions";
+
+const initialState = {
+  title: "Title Of Your Todo List",
+  todos: [
+    {
+      id: 1,
+      value: 'Make a list',
+      completed: false
+    },
+    { 
+        id: 2, 
+        value: 'Make it functional', 
+        completed: false 
+    },
+    { 
+        id: 3, 
+        value: "Make it pretty", 
+        completed: false }
+  ]
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case UPDATE_TITLE:
+      return {
+        ...state,
+        title: action.payload
+      };
+    case ADD_NEW_TODO:
+      return {
+        ...state,
+        todos: state.todos.concat({
+          id: Date.now(),
+          value: action.payload,
+          completed: false
+        })
+      };
+    case TOGGLE_TODO:
```

Comment:
Good job on the toggling here. Once you master immutibility; you're on youre way to mastering redux